pmd-core/src/test/java/net/sourceforge/pmd/lang/rule/xpath/SaxonXPathRuleQueryTest.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.rule.xpath;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import net.sourceforge.pmd.RuleContext;
+import net.sourceforge.pmd.lang.ast.DummyNode;
+import net.sourceforge.pmd.lang.ast.Node;
+import net.sourceforge.pmd.properties.PropertyDescriptor;
+
+public class SaxonXPathRuleQueryTest {
```

Comment:
I think we should Jaxen reacts to list attributes too. I assume, they use the toString of the list, but maybe List attributes should be excluded from Jaxen altogether (like we do for multi-value properties).

Reply:
I've added a unit test. It is indeed the simple String representation of the List and you can only compare the complete list - unless you use `contains()` to check for substrings.

Reply:
IMO leaving that in introduces yet another incompatibility between Jaxen and Saxon.. it may be wise to make those attributes available only with Saxon. I reckon, this could entail adding a method to attribute like `boolean supportsXPath1()` and filtering those in the document navigator.

Maybe we can leave that as is for this release, but clearly advertise in the release notes that using those attributes with XPath 1.0 is experimental (at most) and may be broken by us shortly.

Reply:
We are going to remove Jaxen / XPath 1.0 in 7.0.0 anyway, so I'd not worry so much.

Reply:
The point is that it will be harder for users to transition if we introduce new incompatibilities...

Reply:
I've changed the Jaxen implementation, so that the list attributes are not available.
See e0b0e6330de0616b9452ff6d3f115f3c7fba9f7d

If someone wants to use these list attribute, the XPath needs to use XPath 2.0 now.